examples/ProgrammableButtons/ProgrammableButtons.ino

```diff
@@ -0,0 +1,63 @@
+/*
+  Programmed External Keys (Keyboard)
+
+  For native USB Arduino boards (e.g., Leonardo, Micro, MKR, Nano 33 IoT, Zero, Due)
+
+  When the first button is pressed, the Ctrl + C keyboard shortcut (copy) is emulated.
+  When the second button is pressed, the Ctrl + V keyboard shortcut (paste) is emulated.
+
+  The circuit:
+  The push buttons have 2 connectors.
+  Connect one to GND and the other to the I/O pin defined in the btnPin array.
+
+  created 18 March 2020
+  by Prathamesh Sahasrabhojane (TheShubham99)
+*/
+
+#include <Keyboard.h>
+
+const int btnPin[] = {2, 3, 4, 5};
+int pincount = 2;
+int btnState[pincount];
+int prevBtnState[] = {HIGH, HIGH, HIGH, HIGH};
+
+unsigned long lastDebounceTime[] = {0, 0, 0, 0};
+unsigned long debounceDelay = 50;
+
+void setup() {
+  for (int thisPin = pincount - 1; thisPin >= 0; thisPin--) {
+    pinMode(btnPin[thisPin], INPUT_PULLUP);
+  }
+  Keyboard.begin();
+}
+
+// Place the actions needed after pin press here
+void outputAction(int currentButton) {
+  if (currentButton == 1) {
+    Keyboard.press(KEY_LEFT_CTRL);
+    Keyboard.press('c');
+    delay(100);
+    Keyboard.releaseAll();
+  }
+  else if (currentButton == 2) {
+    Keyboard.press(KEY_LEFT_CTRL);
+    Keyboard.press('v');
+    delay(100);
+    Keyboard.releaseAll();
+  }
+}
+
+void loop() {
+  for (int thisPin = pincount - 1; thisPin >= 0; thisPin--) {
+    btnState[thisPin] = digitalRead(btnPin[thisPin]);
+
+    if ((btnState[thisPin] != prevBtnState[thisPin]) && (btnState[thisPin] == HIGH)) {
```

Comment:
```suggestion
    if ((btnState[thisPin] != prevBtnState[thisPin]) && (btnState[thisPin] == LOW)) {
```
Although the previous code does work, it triggers when the button is released. It is standard for a trigger to occur when the button is pushed, not released. Thus `LOW` is the better choice in this line.

Reply:
I got your point and Thank you very much for the suggestion. :)

But I wanted to ask that as this example is for beginners & in all Arduino examples, the trigger is on when key Reading is HIGH.

Also, the following instructions are from the Arduino official page, where all the beginners will learn.
> When the pushbutton is open (unpressed) there is no connection between the two legs of the pushbutton, so the pin is connected to ground (through the pull-down resistor) and we read a LOW. When the button is closed (pressed), it makes a connection between its two legs, connecting the pin to 5 volts, so that we read a HIGH.

[Link](https://www.arduino.cc/en/tutorial/button)

That is why I think we should keep this statement as it is.
```suggestion
    if ((btnState[thisPin] != prevBtnState[thisPin]) && (btnState[thisPin] == HIGH)) {
```

Please let me know if it is ok/or not?

Reply:
>in all Arduino examples, the trigger is on when key Reading is HIGH.

Wrong. That is correct when you have a button connected between the IO pin and Vcc, with an external pull-down resistor as in the [Button tutorial](https://www.arduino.cc/en/Tutorial/Button). I guess they thought it would be more clear for beginners to have an external resistor and the HIGH == pressed logic.

However, there are plenty of examples that use the opposite logic. That's exactly the circuit you specified in the comment and the code. You specified that the button should be connected between the pin and ground, and you set the internal pull-up resistor (in the original code by using `digitalWrite` to set the pin `HIGH` after setting the pin mode to `INPUT`). In this style of circuit, the pin will be LOW when the button is pressed, as explained in this tutorial:
https://www.arduino.cc/en/Tutorial/InputPullupSerial

For any more advanced example like this one, it's best to use the internal pull-up resistor approach, as you have, because that circuit is easier to build.

Reply:
@per1234 **Thank you for clearing my doubt.

Made the changes. Please review it again :)**